src/goto-programs/string_refine_preprocess.cpp

```diff
@@ -0,0 +1,1343 @@
+/*******************************************************************\
+
+Module: Preprocess a goto-programs so that calls to the java String
+        library are recognized by the string solver
+
+Author: Romain Brenguier
+
+Date:   September 2016
+
+\*******************************************************************/
+
+#include "string_refine_preprocess.h"
+
+#include <util/std_expr.h>
+#include <util/symbol_table.h>
+#include <util/pointer_offset_size.h>
+#include <util/prefix.h>
+#include <goto-programs/class_identifier.h>
+#include <solvers/refinement/string_expr.h>
+
+/*******************************************************************\
+
+Function: string_refine_preprocesst::new_tmp_symbol
+
+  Inputs: a name and a type
+
+ Outputs: a new symbol
+
+ Purpose: add a temporary symbol with the given name and type to the
+          symbol table and returns it
+
+\*******************************************************************/
+
+symbol_exprt string_refine_preprocesst::new_tmp_symbol(
+  const std::string &prefix, const typet &type)
+{
+  auxiliary_symbolt tmp_symbol;
+  std::ostringstream buf;
+  buf << "string_preprocess_tmp" << (next_symbol_id++) << "#" << prefix;
+  std::string name=buf.str();
+  tmp_symbol.base_name=name;
+  tmp_symbol.is_static_lifetime=false;
+  tmp_symbol.mode=ID_java;
+  tmp_symbol.name=name;
+  tmp_symbol.type=type;
+  symbol_table.add(tmp_symbol);
+  return symbol_exprt(name, type);
+}
+
+/*******************************************************************\
+
+Function: string_refine_preprocesst::new_symbol
+
+  Inputs: a name and a type
+
+ Outputs: a new symbol
+
+ Purpose: add a symbol with the given name and type to the symbol table
+          and returns it
+
+\*******************************************************************/
+
+symbol_exprt string_refine_preprocesst::new_symbol(
+  const std::string &prefix, const typet &type, bool static_lifetime)
+{
+  symbolt tmp_symbol;
+  std::ostringstream buf;
+  buf << "string_preprocess" << (next_symbol_id++) << "#" << prefix;
```

Comment:
I used `$` instead (https://github.com/diffblue/cbmc/pull/378/commits/8196d90e4e00009c45d346a92c74dabf60e6a0e9), it seems that's what #523 uses